Enable prometheus via micrometer

kahowell opened the pull request:
Also, added a hack to populate URI in requests.

After merging, it's possible to see metrics by requesting `/actuator/prometheus`. We get already a lot of statistics for free this way, but future work could use methods described in https://github.com/prometheus/client_java/blob/923d23e33135044b59d1d736ebdfd29154ecca02/README.md#custom-collectors to collect more application-specific information.

awood commented on build.gradle:
```diff
@@ -45,10 +45,13 @@ dependencies {
     compile "javax.servlet:javax.servlet-api:3.1.0"
     compile "org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.1_spec:1.0.2.Final"
     compile "org.springframework.boot:spring-boot-autoconfigure:$spring_boot_version"
+    compile "org.springframework.boot:spring-boot-starter-actuator:$spring_boot_version"
```
If we're adding one of the starters, we should look at our dependency graph and remove duplicates.  I imagine that the starter pulls in `spring-boot-autoconfigure`, `spring-beans`, and `spring-context`.

Personally, I'm torn on whether to use the starters or not.  They do drag in **a lot** of extra dependencies, but they make life easier.

kahowell replied:
Ack. Will take a look at whether I can just pull in a couple deps instead.

kahowell commented:
@awood, I rebased, and went ahead and replaced spring-boot-starter-actuator with spring-boot-actuator-autoconfigure, and that was all that was necessary. :smile:

kahowell commented:
Just did one last force-push, went ahead and updated copyright header while I was in there.
